File: android/DFM/src/main/java/com/dontflymoney/baseactivity/FixOrientationActivity.java
package com.dontflymoney.baseactivity;

import android.app.Activity;
import android.content.pm.ActivityInfo;
import android.os.Bundle;

public class FixOrientationActivity extends Activity
{
	private static int oldConfigInt;
	protected boolean rotated;

	@Override
	protected void onCreate(Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);
		
		rotated =
			(oldConfigInt & ActivityInfo.CONFIG_ORIENTATION)
			==
			ActivityInfo.CONFIG_ORIENTATION;
	}
	
	@Override
	protected void onDestroy()
	{
		super.onDestroy();
		oldConfigInt = getChangingConfigurations();
	}
	
	@Override
	protected void onResume()
	{
		super.onResume();
		oldConfigInt = 0;
	}
	
}

File: android/DFM/src/main/kotlin/com/darakeon/dfm/activities/base/FixOrientationActivity.kt
package com.darakeon.dfm.activities.base

import android.app.Activity
import android.content.pm.ActivityInfo
import android.os.Bundle

open class FixOrientationActivity : Activity() {
    protected var rotated: Boolean = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        rotated = oldConfigInt and ActivityInfo.CONFIG_ORIENTATION == ActivityInfo.CONFIG_ORIENTATION
    }

    override fun onDestroy() {
        super.onDestroy()
        oldConfigInt = changingConfigurations
    }

    override fun onResume() {
        super.onResume()
        oldConfigInt = 0
    }

    companion object {
        private var oldConfigInt: Int = 0
    }

}
